src/server/scripts/Northrend/zone_sholazar_basin.cpp

```diff
@@ -772,6 +772,162 @@ class spell_shango_tracks : public SpellScriptLoader
     }
 };
 
+enum TheGreatHuntersChallenge
+{
+    QUEST_THE_GREAT_HUNTERS_CHALLENGE = 12592,
+    NPC_SUMMONED_DROSTAN              = 28857,
+    SPELL_SUMMON_DROSTAN              = 52556,
+    SPELL_DROSTAN_AURA_SLEEP          = 55474,
+
+    EVENT_DROSTAN_RUN_AWAY     = 1,
+    EVENT_DROSTAN_SLEEP        = 2,
+    EVENT_DROSTAN_SAY_7        = 3,
+    EVENT_DROSTAN_OBJECTIVE_6  = 6,
+    EVENT_DROSTAN_OBJECTIVE_11 = 11,
+    EVENT_DROSTAN_OBJECTIVE_21 = 21,
+    EVENT_DROSTAN_OBJECTIVE_28 = 28,
+    EVENT_DROSTAN_OBJECTIVE_35 = 35,
+    EVENT_DROSTAN_OBJECTIVE_41 = 41,
+    EVENT_DROSTAN_OBJECTIVE_49 = 49,
+    EVENT_DROSTAN_OBJECTIVE_56 = 56,
+
+    DROSTAN_SAY_2  = 2,
+    DROSTAN_SAY_3  = 3,
+    DROSTAN_SAY_4  = 4,
+    DROSTAN_SAY_5  = 5,
+    DROSTAN_SAY_6  = 6,
+    DROSTAN_SAY_7  = 7,
+    DROSTAN_SAY_8  = 8,
+    DROSTAN_SAY_9  = 9,
+    DROSTAN_SAY_10 = 10,
+
+};
+
+class npc_drostan : public CreatureScript
+{
+public:
+    npc_drostan() : CreatureScript("npc_drostan") { }
+
+    struct npc_drostanAI : public CreatureAI
+    {
+        npc_drostanAI(Creature* creature) : CreatureAI(creature) { }
+
+        void CallEvent(Player* player, uint32 progress)
+        {
+            _playerGUID = player->GetGUID();
+            _events.ScheduleEvent(progress, Seconds(3));
+        }
+
+        void UpdateAI(uint32 diff)
+        {
+            _events.Update(diff);
+
+            if (uint32 eventId = _events.ExecuteEvent())
+            {
+                Unit* player = ObjectAccessor::GetUnit(*me, _playerGUID);
+                if (!player)
+                    return;
+
+                switch (eventId)
+                {
+                    case EVENT_DROSTAN_RUN_AWAY:
+                        me->GetMotionMaster()->MovePoint(0, me->GetPosition().GetPositionWithOffset(Position(0.f, 20.f, 0.f, 0.f)), true);
```

Comment:
does the creature jump between events as different players reach different progress ? should only 1 player at a time affect the progress ?

Reply:
Nope, each spawn does exactly one event (based on how many creatures you killed, they trigger at the 6/11/21/28/35/41/49/56th kill) and then despawns.